hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -192,6 +237,158 @@ singlePartyCommitsFromExternal tracer workDir node@RunningNode{networkId} hydraS
+      scriptUtxo1 <- createScriptOutput pparams scriptAddress1 someSk normalUTxO1 datum
+      normalUTxO2 <- seedFromFaucet node someVk 10_000_000 Normal (contramap FromFaucet tracer)
+      scriptUtxo2 <- createScriptOutput pparams scriptAddress2 someSk normalUTxO2 datum
+      let scriptUtxosInfo = [DraftUTxO scriptUtxo1 (Just scriptInfo1)] <> [DraftUTxO scriptUtxo2 (Just scriptInfo2)]
+      -- Request to build a draft commit tx from hydra-node
+      regularUtxo1 <- seedFromFaucet node someVk 10_000_000 Normal (contramap FromFaucet tracer)
+      regularUtxo2 <- seedFromFaucet node someVk 10_000_000 Normal (contramap FromFaucet tracer)
+      let regularUtxosInfo = [DraftUTxO regularUtxo1 Nothing] <> [DraftUTxO regularUtxo2 Nothing]
+      let clientPayload = DraftCommitTxRequest @Tx (regularUtxosInfo <> scriptUtxosInfo)
+      response <-
+        runReq defaultHttpConfig $
+          req
+            POST
+            (http "127.0.0.1" /: "commit")
+            (ReqBodyJson clientPayload)
+            (Proxy :: Proxy (JsonResponse (DraftCommitTxResponse Tx)))
+            (port $ 4000 + hydraNodeId)
+      responseStatusCode response `shouldBe` 200
+      let DraftCommitTxResponse commitTx = responseBody response
+      let signedCommitTx = signWith someSk commitTx
+      submitTransaction networkId nodeSocket signedCommitTx
+
+      waitFor tracer 60 [n1] $
+        output "HeadIsOpen" ["utxo" .= (regularUtxo1 <> regularUtxo2), "headId" .= headId]
+ where
+  RunningNode{networkId, nodeSocket} = node
+
+  -- TODO: refactor into simpler
+  createScriptOutput ::
+    ToScriptData a =>
+    ProtocolParameters ->
+    AddressInEra ->
+    SigningKey PaymentKey ->
+    UTxO ->
+    a ->
+    IO UTxO
+  createScriptOutput pparams scriptAddress sk utxo datum = do
+    let outputs = [scriptTxOut]
+        totalDeposit = sum (selectLovelace . txOutValue <$> outputs)
+        someUTxO =
+          maybe mempty UTxO.singleton $
+            UTxO.find (\o -> selectLovelace (txOutValue o) > totalDeposit) utxo
+    buildTransaction
+      networkId
+      nodeSocket
+      changeAddress
+      someUTxO
+      collateralTxIns
+      outputs
+      >>= \case
+        Left e ->
+          throwErrorAsException e
+        Right body -> do
+          let tx = makeSignedTransaction [makeShelleyKeyWitness body (WitnessPaymentKey sk)] body
+          submitTransaction networkId nodeSocket tx
+          newUtxo <- awaitTransaction networkId nodeSocket tx
+          let scriptUtxo = UTxO.filter (\out -> txOutAddress out == scriptAddress) newUtxo
+          pure scriptUtxo
+   where
+    collateralTxIns = mempty
+
+    vk = getVerificationKey sk
+
+    changeAddress = mkVkAddress networkId vk
+
+    scriptTxOut =
+      mkTxOutAutoBalance
+        pparams
+        scriptAddress
+        mempty
+        (mkTxOutDatumHash datum)
+        ReferenceScriptNone
+
+singlePartyCantCommitExternallyWalletUtxo ::
```

Comment:
I would make this test a _BehaviorSpec_ test so that there is less plumbing involved and keep only a very basic test here.